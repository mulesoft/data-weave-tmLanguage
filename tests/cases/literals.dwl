%dw 2.0
---
{
  "boolean":{
    "true" : true,
    "false": false
  },
  "Number": {
    "int": 123,
    "decimal": 123.23
  },
  "string": {
    "singleQuote" : 'A String',
    "backtickQuote" : `A String`,
    "doubleQuote" : "A String"
  },
  "regex": /foo/,
}