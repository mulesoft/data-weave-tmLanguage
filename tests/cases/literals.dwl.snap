>%dw 2.0
#^^^ source.data-weave meta.directive.version.dw keyword.directive.version.dw
#   ^ source.data-weave meta.directive.version.dw
#    ^^^ source.data-weave meta.directive.version.dw constant.numeric.version.dw
>---
#^^^ source.data-weave keyword.operator.body-marker.dw
>{
#^^ source.data-weave
>  "boolean":{
#^^ source.data-weave
#  ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#   ^^^^^^^ source.data-weave string.quoted.double.dw
#          ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#           ^ source.data-weave keyword.operator.declaration.dw
#            ^^ source.data-weave
>    "true" : true,
#^^^^ source.data-weave
#    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#     ^^^^ source.data-weave string.quoted.double.dw
#         ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#          ^ source.data-weave
#           ^ source.data-weave keyword.operator.declaration.dw
#            ^ source.data-weave
#             ^^^^ source.data-weave constant.language.dw
#                 ^^ source.data-weave
>    "false": false
#^^^^ source.data-weave
#    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#     ^^^^^ source.data-weave string.quoted.double.dw
#          ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#           ^ source.data-weave keyword.operator.declaration.dw
#            ^ source.data-weave
#             ^^^^^ source.data-weave constant.language.dw
>  },
#^^^^^ source.data-weave
>  "Number": {
#^^ source.data-weave
#  ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#   ^^^^^^ source.data-weave string.quoted.double.dw
#         ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#          ^ source.data-weave keyword.operator.declaration.dw
#           ^^^ source.data-weave
>    "int": 123,
#^^^^ source.data-weave
#    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#     ^^^ source.data-weave string.quoted.double.dw
#        ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#         ^ source.data-weave keyword.operator.declaration.dw
#          ^ source.data-weave
#           ^^^ source.data-weave constant.numeric.dw
#              ^^ source.data-weave
>    "decimal": 123.23
#^^^^ source.data-weave
#    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#     ^^^^^^^ source.data-weave string.quoted.double.dw
#            ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#             ^ source.data-weave keyword.operator.declaration.dw
#              ^ source.data-weave
#               ^^^^^^ source.data-weave constant.numeric.dw
>  },
#^^^^^ source.data-weave
>  "string": {
#^^ source.data-weave
#  ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#   ^^^^^^ source.data-weave string.quoted.double.dw
#         ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#          ^ source.data-weave keyword.operator.declaration.dw
#           ^^^ source.data-weave
>    "singleQuote" : 'A String',
#^^^^ source.data-weave
#    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#     ^^^^^^^^^^^ source.data-weave string.quoted.double.dw
#                ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#                 ^ source.data-weave
#                  ^ source.data-weave keyword.operator.declaration.dw
#                   ^ source.data-weave
#                    ^ source.data-weave string.quoted.single.dw punctuation.definition.string.begin.dw
#                     ^^^^^^^^ source.data-weave string.quoted.single.dw
#                             ^ source.data-weave string.quoted.single.dw punctuation.definition.string.end.dw
#                              ^^ source.data-weave
>    "backtickQuote" : `A String`,
#^^^^ source.data-weave
#    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#     ^^^^^^^^^^^^^ source.data-weave string.quoted.double.dw
#                  ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#                   ^ source.data-weave
#                    ^ source.data-weave keyword.operator.declaration.dw
#                     ^ source.data-weave
#                      ^ source.data-weave string.quoted.backtick.dw punctuation.definition.string.begin.dw
#                       ^^^^^^^^ source.data-weave string.quoted.backtick.dw
#                               ^ source.data-weave string.quoted.backtick.dw punctuation.definition.string.end.dw
#                                ^^ source.data-weave
>    "doubleQuote" : "A String"
#^^^^ source.data-weave
#    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#     ^^^^^^^^^^^ source.data-weave string.quoted.double.dw
#                ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#                 ^ source.data-weave
#                  ^ source.data-weave keyword.operator.declaration.dw
#                   ^ source.data-weave
#                    ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#                     ^^^^^^^^ source.data-weave string.quoted.double.dw
#                             ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
>  },
#^^^^^ source.data-weave
>  "regex": /foo/,
#^^ source.data-weave
#  ^ source.data-weave string.quoted.double.dw punctuation.definition.string.begin.dw
#   ^^^^^ source.data-weave string.quoted.double.dw
#        ^ source.data-weave string.quoted.double.dw punctuation.definition.string.end.dw
#         ^ source.data-weave keyword.operator.declaration.dw
#          ^ source.data-weave
#           ^^^^^ source.data-weave string.regexp.dw
#                ^^ source.data-weave
>}
#^^ source.data-weave